packages/form-core/src/registration/FormRegisteringMixin.js

```diff
@@ -32,28 +33,27 @@ export const FormRegisteringMixin = dedupeMixin(
         this._unregisterFormElement();
       }
 
-      __setupRegistrationHook() {
-        if (formRegistrarManager.ready) {
-          this._dispatchRegistration();
-        } else {
-          formRegistrarManager.addEventListener(
-            'all-forms-open-for-registration',
-            this.__boundDispatchRegistration,
-          );
+      _dispatchRegistration() {
+        let commentNode;
+        if (window.ShadyDOM) {
```

Comment:
since the commentNode is not cleaned up on unregister, maybe prevent creating new dispatcher commentNodes when elements are being moved around

Reply:
commentNode is always removed directly after dispatching the event